### PR TITLE
feat: support façade-handling properties.

### LINKED ISSUE #206: Add support for properties for facade converter
JS has `Map.size` while Dart has `Map.length`.

It seems like properties are not supported, only methods, right ?

/cc @mprobst

### PR DESCRIPTION
**Pull request by mprobst**
Fixes #206.

### CODE REVIEW

File: lib/facade_converter.ts [hunk 1]
```diff
@@ -3,22 +3,27 @@ import base = require('./base');
...
-          .filter((k) => this.subs[fileName].hasOwnProperty(k))
+    this.extractPropertyNames(this.callHandlers);
+    this.extractPropertyNames(this.propertyHandlers);
+  }
+
+  private extractPropertyNames(m: ts.Map<ts.Map<any>>) {
+    for (var fileName in m) {
+      Object.keys(m[fileName])
+          .filter((k) => m[fileName].hasOwnProperty(k))
           .map((fnName) => fnName.substring(fnName.lastIndexOf('.') + 1))
```
Comment (vicb) [inline lib/facade_converter.ts]: nit: fnName -> propName ?

File: lib/facade_converter.ts [hunk 2]
```diff
@@ -190,34 +233,13 @@ export class FacadeConverter extends base.TranspilerBase {
...
-
-  private isConstCall(node: ts.CallExpression): boolean {
-    return node && base.ident(node.expression) === 'CONST_EXPR';
-  }
+  private propertyHandlers: ts.Map<ts.Map<PropertyHandler>> = {
+    'angular2/traceur-runtime': {
+      'Map.size': (p: ts.PropertyAccessExpression) => {
+        this.visit(p.expression);
+        this.emit('.');
+        this.emit('length');
```
Comment (vicb) [inline lib/facade_converter.ts]: :+1:

File: lib/expression.ts [hunk 3]
```diff
@@ -68,12 +68,15 @@ class ExpressionTranspiler extends base.TranspilerBase {
 
       case ts.SyntaxKind.PropertyAccessExpression:
         var propAccess = <ts.PropertyAccessExpression>node;
-        this.visit(propAccess.expression);
         if (propAccess.name.text === 'stack' &&
             this.hasAncestor(propAccess, ts.SyntaxKind.CatchClause)) {
+          // Handle `e.stack` accesses in catch clauses by mangling to `e_stack`.
+          // FIXME: Use type checker/FacadeConverter to make sure this is actually Error.stack.
+          this.visit(propAccess.expression);
           this.emitNoSpace('_stack');
```
Comment (vicb) [inline lib/expression.ts]: - what about adding a test here ?
- could this be solved by adding a transformer that visits the catch clause ? We should be able to determine if `.stack` is accessed on the exception ?

File: lib/facade_converter.ts (hunk 1 above)
Comment (mprobst) [inline lib/facade_converter.ts]: Done.

**Comment by vicb** [comment]
a few comments, otherwise LGTM

File: lib/expression.ts (hunk 3 above)
Comment (mprobst) [inline lib/expression.ts]: There is a test for this behaviour (in `test/statement_test.ts`), note that I'm only rewording the comment and slightly refactoring the code.

**Comment by mprobst** [comment]
Thanks for the review!

**Pull request merged**